src/GHC/SourceGen/Decl.hs

```diff
@@ -308,8 +326,16 @@ data' = newOrDataType DataType
 -- > =====
 -- > conDecl "Foo" [field (var "a"), field (var "Int")]
 prefixCon :: OccNameStr -> [Field] -> ConDecl'
-prefixCon name fields = renderCon98Decl name
-    $ PrefixCon $ map renderField fields
+prefixCon name fields =
+    renderCon98Decl
+        name
+            $ PrefixCon
+#if MIN_VERSION_ghc(9,0,1)
+                $ map (HsScaled (HsUnrestrictedArrow NormalSyntax) . renderField)
```

Comment:
`HsConDetails` of `HsConDeclDetails` now yas `HsScaled` wrapped around its first field. I've stubbed this with `(HsScaled (HsUnrestrictedArrow NormalSyntax)`